### PR TITLE
Vacms 7082 product api additions

### PR DESCRIPTION
**Pull request by timcosgrove**
## Description

Relates to #7082

### CODE REVIEW

File: docroot/modules/custom/va_gov_api/src/Resource/ResourcesByTag.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+use Drupal\jsonapi\ResourceType\ResourceType;
+use Drupal\node\NodeInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Resource for returning a collection of 'resource' product nodes.
+ *
+ * Apologies for the naming collision. Currently limited to q_a nodes.
+ */
+class ResourcesByTag extends VaGovApiEntityResourceBase {
```
Comment (timcosgrove) [inline docroot/modules/custom/va_gov_api/src/Resource/ResourcesByTag.php:16]: This one is in process, don't review it yet.

File: docroot/modules/custom/va_gov_api/src/Resource/BannerAlerts.php [hunk 2]
```diff
@@ -2,155 +2,261 @@
...
-   * The entity type manager.
-   *
-   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   * {@inheritDoc}
    */
-  protected $entityTypeManager;
+  protected function collectResourceData(Request $request, ResourceType $resource_type) {
+    // Doing this as a case/switch for now. There are almost certainly better
+    // ways to do this.
+    switch ($resource_type->getTypeName()) {
```
Comment (indytechcook) [inline docroot/modules/custom/va_gov_api/src/Resource/BannerAlerts.php:23]: > / Doing this as a case/switch for now. There are almost certainly better
> // ways to do this.

Yep, by using Plugins!

**Comment by indytechcook** [comment]
Just some thoughts

- Make sure to type hint returns and make the responses as consistent as possible
- Separate out the builder of the Resource from from the actual Resource object.
- Create a consistent interface to pull the data from different resource types
- Use the Drupal Plugin API for the different builders for the resource types
- The plugin used will depend on the resource type.
- The plugin could also provide the path as part of it's metadata instead of a bunch of routing entries. https://www.drupal.org/node/2122201
- Try to follow a Composition pattern over inheritance (Single resource class with the different plugins to get the data is one example)

Ideally we need someway to integrate with the params to build the query using the filters, pages, etc in a way which is reused across all request.  I would assume this exists already since the resource module does this.

File: docroot/modules/custom/va_gov_api/src/Resource/BannerAlerts.php (hunk 2 above)
Comment (timcosgrove) [reply docroot/modules/custom/va_gov_api/src/Resource/BannerAlerts.php:23]: Want to chat about this at some point.

File: docroot/modules/custom/va_gov_api/src/Resource/QandA.php [hunk 3]
```diff
@@ -3,83 +3,87 @@
 namespace Drupal\va_gov_api\Resource;
```
Comment (timcosgrove) [inline docroot/modules/custom/va_gov_api/src/Resource/QandA.php:3]: This whole thing - retrieving an entity with a custom endpoint - is almost certainly not something we would actually do.

**Review by indytechcook — APPROVED** [review]

**Pull request merged**